sites/hashdev/src/_pages/blog/4_announcement-error-stack.mdx

```diff
@@ -0,0 +1,285 @@
+
+However, we've purposefully added in some requirements that result in a little bit of extra upfront **development** overhead. We think having explicitly typed `Result` values is worth requiring, and especially helpful in identifying different types across module/crate boundaries, so we still want users to create types (`Context`s). This approach makes `error-stack` usable not only for binaries but also for libraries. In most cases, the caller of a library only cares about the error directly returned from a library, not an underlying OS error. As the current context is _always_ known, the user directly has all type information without optimistically trying to downcast to an error type (which remains possible). This also implies that **more times than not** a developer is _forced_ to add a new context because the type system requires it. We've found from experience that this automatically greatly improves error reporting.
+
+We haven't made it easy to create a new `Report` from a `String` because it requires the type to implement `Context`. It's still possible to use a newtype `UniqueError(String)` if a user wants to, but we discourage this usage because we've found that a lot of those places benefit from attaching a string message on a well-defined context instead. In any case, other libraries most probably still have to change the context from `UniqueError` to their context because of the explicit type requirement.
+
+**By consciously imposing more restrictions in some areas we hope to reduce the _overall_ cost of development, in keeping with the philosophy underpinning Rust itself.**
+
+## Some other cool things about `error-stack` that we don't have time for!
+
+We don't have space here to talk about all of the features of `error-stack`, but here are a few more teasers. [Check out the repository](https://github.com/hashintel/hash/tree/main/packages/libs/error-stack) for even more information:
+
+- The complete crate is written for `no-std` environments. However, when using `std`, a blanket implementation for `Context` for any `Error` is provided
+- The blanket implementation for `Error` also makes the library compatible with almost all other libraries using the official trait
+  - This means that if you pick this up, you should be able to migrate gradually (like we're doing with hEngine)
+- Just like [`eyre`](https://crates.io/crates/eyre), the crate is able to set its own `Display` and `Debug` hooks that are called when printing the `Report`, which means you can write your own custom formatting and do other cool things
+- We have in-built support for backtraces on the nightly channel and feature-flagged support for `tracing` to capture a `SpanTrace`
+
+## The Future of `error-stack`
```

Comment:
```suggestion
### The Future of `error-stack`
```

Reply:
Closing this suggestion in favor of https://github.com/hashintel/hash/pull/628/files#r894363610